pkg/cloudevents/transport/http/transport.go

```diff
@@ -170,7 +170,9 @@ func (t *Transport) Send(ctx context.Context, event cloudevents.Event) (context.
 func (t *Transport) obsSend(ctx context.Context, event cloudevents.Event) (context.Context, *cloudevents.Event, error) {
 	if t.Client == nil {
 		t.crMu.Lock()
-		t.Client = &http.Client{}
+		if t.Client == nil {
```

Comment:
this code can only get here if `t.Client == nil`, see line 171

Reply:
however, this is not thread safe. It there are two goroutines exec this func concurrently, the first create the t.Client at line 173, and the other is blocked at line 172. Later on the first goroutine release the lock  at line 174, and the second goroutine will get the lock and create the t.Client again at line 173.

Reply:
ah I see, can you kill line 171?

Reply:
well... nevermind, no need to toggle that lock over and over....